### PR TITLE
`Forms` : Merge Feature branches/forms into v.next for 200.6

### PR DESCRIPTION
**Pull request by kaushikrw**
Related to issue: #[kotlin/4954](https://devtopia.esri.com/runtime/kotlin/issues/4954)

### Description:

Merge the latest feature form development into `v.next` for the 200.6 release.

### Summary of changes:

- Introduced a new parameter `onBarcodeButtonClicked` which allows users to use a custom barcode scanner. This change required a new constructor while deprecating the existing ones to preserve binary compatibility. #639 
- Added Tests for the above addition.
- Added validation error messages for GUID field types.
- Fixed a lint warning where some data classes had an internal ctor but exposed the public `copy` method. These data classes ctor is made public. #642 
- Fixed an issue where other `FormElement` types were not rendered within a `GroupFormElement`.
- UI design update for single line and multi line text fields as part of the new UI/UX design review. #665 
- UI design update for barcode fields as part of the new UI/UX design review. #648 

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [x] link: https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/202/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [x] Yes
  - [ ] No

### CODE REVIEW

**Review by sorenoid — APPROVED** [review]

**Comment by kaushikrw** [comment]
The vTest has now completed. The test `testAttachmentsElementTheming` is likely failing due to the data in the underlying map. I will fix the data later.

File: toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/internal/components/barcode/BarcodeTextFieldDefaults.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+
+/**
+ * Default values for [BarcodeTextField].
+ */
+internal object BarcodeTextFieldDefaults {
+
+    val barcodeIconTintColor : Color
```
Comment (shubham7109) [inline toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/internal/components/barcode/BarcodeTextFieldDefaults.kt:29]:
```suggestion
    val barcodeIconTintColor: Color
```

**Review by shubham7109 — APPROVED** [review]
@kaushikrw The changes in this PR looks good to me 👍,  and I can see a bunch of the changes here have been reviewed previously. 

I took some time to scan through the module and I have some questions on the copyright usage: 

- `ValidationErrorState.kt`, `FieldValidation.kt`, `FormTextFieldNumericTests.kt`, `FormTextFieldRangeNumericTests.kt`, `ImageUtil.kt`, `TestChallengeHandler.kt`,  may be using the wrong internal copyright. Or is it intended, I'm not too sure 😅 
- `CorePrototypes.kt` also uses internal copyright with the message: `This entire file will be removed before the 200.3.0 release.`

**Comment by kaushikrw** [comment]
@shubham7109 Good catch. I have updated the copyrights. As for `CorePrototypes.kt` that was the original intention but it is now just a file of related utility functions.

**Comment by kaushikrw** [comment]
Thanks @shubham7109, @sorenoid. Merging now.

**Pull request merged**